targets/TARGET_STM/stm_spi_api.c

```diff
@@ -92,6 +92,16 @@ void init_spi(spi_t *obj)
     }
 }
 
+SPIName spi_get_peripheral_name(PinName mosi, PinName miso, PinName sclk) {
+    SPIName spi_mosi = (SPIName)pinmap_peripheral(mosi, PinMap_SPI_MOSI);
+    SPIName spi_miso = (SPIName)pinmap_peripheral(miso, PinMap_SPI_MISO);
```

Comment:
In case of "3 wires SPI" , miso is set as "NC" - so this needs to be taken into account.
Please add check, if miso is NC, only merge mosi and sclk

Reply:
pinmap_merge should handle a situation when other pin is set as NC, but I'll provide also a check here.

Reply:
Done

Reply:
I think this needs more change. The check should happen before calling to   
`SPIName spi_miso = (SPIName)pinmap_peripheral(miso, PinMap_SPI_MISO);`

Otherwise, the system would crash here:
https://github.com/ARMmbed/mbed-os/blob/master/hal/mbed_pinmap_common.c#L78

Reply:
If miso is NC, should it even get there?
https://github.com/ARMmbed/mbed-os/blob/master/hal/mbed_pinmap_common.c#L73

Reply:
You're right - so that's OK - thanks !